test_remote_storage_upload_queue_retries is flaky

```
2022-12-09T16:04:38.3224170Z =================================== FAILURES ===================================
2022-12-09T16:04:38.3224482Z ______________ test_remote_storage_upload_queue_retries[local_fs] ______________
2022-12-09T16:04:38.3225083Z [gw0] linux -- Python 3.9.2 /github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/bin/python
2022-12-09T16:04:38.3225585Z test_runner/regress/test_remote_storage.py:314: in test_remote_storage_upload_queue_retries
2022-12-09T16:04:38.3225982Z     assert not churn_while_failpoints_active_thread.is_alive()
2022-12-09T16:04:38.3226239Z E   assert not True
2022-12-09T16:04:38.3226656Z E    +  where True = <bound method Thread.is_alive of <Thread(Thread-3, started 140135066326784)>>()
2022-12-09T16:04:38.3227232Z E    +    where <bound method Thread.is_alive of <Thread(Thread-3, started 140135066326784)>> = <Thread(Thread-3, started 140135066326784)>.is_alive
```

https://github.com/neondatabase/neon/actions/runs/3658686712/jobs/6184058965

Last 10 failures found in grafana for example [this](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-6872/8013489918/index.html#testresult/8235938280e0f836/retries) or [this](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-6885/8011078488/index.html#testresult/9d3f442e027aec01/retries) all hit the same assertion:

```
test_runner/regress/test_remote_storage.py:290: in test_remote_storage_upload_queue_retries
    assert gc_result["layers_removed"] > 0
E   assert 0 > 0
```

--- a/test_runner/regress/test_remote_storage.py
+++ b/test_runner/regress/test_remote_storage.py
@@ -256,7 +256,7 @@ def test_remote_storage_upload_queue_retries(
             ]
         )
 
-    FOO_ROWS_COUNT = 10000
+    FOO_ROWS_COUNT = 4000
 
     def overwrite_data_and_wait_for_it_to_arrive_at_pageserver(data):
         # create initial set of layers & upload them with failpoints configured

--- a/test_runner/regress/test_remote_storage.py
+++ b/test_runner/regress/test_remote_storage.py
@@ -228,9 +228,9 @@ def test_remote_storage_upload_queue_retries(
     tenant_id, timeline_id = env.neon_cli.create_tenant(
         conf={
             # small checkpointing and compaction targets to ensure we generate many upload operations
-            "checkpoint_distance": f"{128 * 1024}",
+            "checkpoint_distance": f"{64 * 1024}",
             "compaction_threshold": "1",
-            "compaction_target_size": f"{128 * 1024}",
+            "compaction_target_size": f"{64 * 1024}",
             # no PITR horizon, we specify the horizon when we request on-demand GC
             "pitr_interval": "0s",
             # disable background compaction and GC. We invoke it manually when we want it to happen.
@@ -256,21 +256,24 @@ def test_remote_storage_upload_queue_retries(
             ]
         )
 
+    FOO_ROWS_COUNT = 10000
+
     def overwrite_data_and_wait_for_it_to_arrive_at_pageserver(data):
         # create initial set of layers & upload them with failpoints configured
-        endpoint.safe_psql_many(
-            [
-                f"""
-               INSERT INTO foo (id, val)
-               SELECT g, '{data}'
-               FROM generate_series(1, 20000) g
-               ON CONFLICT (id) DO UPDATE
-               SET val = EXCLUDED.val
-               """,
-                # to ensure that GC can actually remove some layers
-                "VACUUM foo",
-            ]
-        )
+        for _v in range(2):
+            endpoint.safe_psql_many(
+                [
+                    f"""
+                    INSERT INTO foo (id, val)
+                    SELECT g, '{data}'
+                    FROM generate_series(1, {FOO_ROWS_COUNT}) g
+                    ON CONFLICT (id) DO UPDATE
+                    SET val = EXCLUDED.val
+                    """,
+                    # to ensure that GC can actually remove some layers
+                    "VACUUM foo",
+                ]
+            )
         wait_for_last_flush_lsn(env, endpoint, tenant_id, timeline_id)
 
     def get_queued_count(file_kind, op_kind):
@@ -333,7 +336,7 @@ def test_remote_storage_upload_queue_retries(
 
     # The churn thread doesn't make progress once it blocks on the first wait_completion() call,
     # so, give it some time to wrap up.
-    churn_while_failpoints_active_thread.join(30)
+    churn_while_failpoints_active_thread.join(60)
     assert not churn_while_failpoints_active_thread.is_alive()
     assert churn_thread_result[0]
 
@@ -365,7 +368,7 @@ def test_remote_storage_upload_queue_retries(
     log.info("restarting postgres to validate")
     endpoint = env.endpoints.create_start("main", tenant_id=tenant_id)
     with endpoint.cursor() as cur:
-        assert query_scalar(cur, "SELECT COUNT(*) FROM foo WHERE val = 'd'") == 20000
+        assert query_scalar(cur, "SELECT COUNT(*) FROM foo WHERE val = 'd'") == FOO_ROWS_COUNT
 
 
 def test_remote_timeline_client_calls_started_metric(

